src/App.js

```diff
@@ -1,16 +1,31 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [likedMessages, setLikedMessages] = useState(0);
+
+  const likeMessage = () => {
+    setLikedMessages(likedMessages => likedMessages + 1);
+  };
+
+  const unlikeMessage = () => {
+    setLikedMessages(likedMessages => likedMessages - 1);
+  };
+  
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>Chat Messages</h1>
+        <h2>{likedMessages} ❤️s</h2>
       </header>
       <main>
-        {/* Wave 01: Render one ChatEntry component
-        Wave 02: Render ChatLog component */}
+        <ChatLog
+          entries={chatMessages}
+          likeMessage={likeMessage}
+          unlikeMessage={unlikeMessage}
```

Comment:
Following my comment above about creating an event handler in app.js to help you keep track of a message's 'liked' status, you would need to pass that event handler from App to ChatLog and finally into ChatEntry.
```javascript
<ChatLog
    entries={chatMessageData}
    updateEntryData={updateEntryData}
/>